include/ofi_mem.h

```diff
@@ -349,21 +349,19 @@ void *util_buf_get_by_index(struct util_buf_pool *pool, size_t index);
 
 static inline void *util_buf_get(struct util_buf_pool *pool)
 {
-	struct slist_entry *entry;
-	entry = slist_remove_head(&pool->buf_list);
-	return entry;
+	return slist_remove_head(&pool->buf_list);
 }
 
 static inline void util_buf_release(struct util_buf_pool *pool, void *buf)
 {
-	union util_buf *util_buf = buf;
-	slist_insert_head(&util_buf->entry, &pool->buf_list);
+	slist_insert_head(&((union util_buf * )buf)->entry, &pool->buf_list);
```

Comment:
The above changes are fine, but I doubt the compiler kept the stack variables anyway.

Reply:
Just to be honest, I didn't check this :)